packages/components/src/date-time/date/index.tsx

```diff
@@ -100,139 +60,284 @@ function DatePickerDay( { day, events = [] }: DatePickerDayProps ) {
+	// pressed. Only do this if focus is allowed - this stops us stealing focus
+	// from e.g. a TimePicker input. Note that isFocusAllowed is not a dep as
+	// there is no point calling focus() on an already focused element.
+	useEffect( () => {
+		if ( ref.current && isFocusable && isFocusAllowed ) {
+			ref.current.focus();
 		}
-
-		return events.filter( ( eventDay ) =>
-			day.isSame( eventDay.date, 'day' )
-		);
-	};
-
-	const momentDate = getMomentDate( currentDate );
+	}, [ isFocusable ] );
 
 	return (
-		<div className="components-datetime__date" ref={ nodeRef }>
-			<DayPickerSingleDateController
-				date={ momentDate }
-				initialVisibleMonth={ null }
-				daySize={ 30 }
-				horizontalMonthPadding={ 0 }
-				focused
-				hideKeyboardShortcutsPanel
-				// This is a hack to force the calendar to update on month or year change
-				// https://github.com/airbnb/react-dates/issues/240#issuecomment-361776665
-				key={ `datepicker-controller-${
-					momentDate ? momentDate.format( 'MM-YYYY' ) : 'null'
-				}` }
-				noBorder
-				numberOfMonths={ 1 }
-				onDateChange={ onChangeMoment }
-				transitionDuration={ 0 }
-				weekDayFormat="ddd"
-				dayAriaLabelFormat={ ARIAL_LABEL_TIME_FORMAT }
-				isRTL={ isRTL() }
-				isOutsideRange={ ( date ) => {
-					return !! isInvalidDate && isInvalidDate( date.toDate() );
-				} }
-				firstDayOfWeek={ startOfWeek }
-				onPrevMonthClick={ onMonthPreviewedHandler }
-				onNextMonthClick={ onMonthPreviewedHandler }
-				renderDayContents={ ( day ) => (
-					<DatePickerDay
-						day={ day }
-						events={ getEventsPerDay( day ) }
-					/>
-				) }
-				renderMonthElement={ ( { month } ) => (
-					<>
-						<strong>{ month.format( 'MMMM' ) }</strong>{ ' ' }
-						{ month.format( 'YYYY' ) }
-					</>
-				) }
-				renderNavPrevButton={ ( { ariaLabel, ...props } ) => (
-					<NavPrevButton
-						icon={ arrowLeft }
-						variant="tertiary"
-						aria-label={ ariaLabel }
-						{ ...props }
-					/>
-				) }
-				renderNavNextButton={ ( { ariaLabel, ...props } ) => (
-					<NavNextButton
-						icon={ arrowRight }
-						variant="tertiary"
-						aria-label={ ariaLabel }
-						{ ...props }
-					/>
-				) }
-				onFocusChange={ noop }
-			/>
-		</div>
+		<DayButton
+			ref={ ref }
+			className="components-datetime__date__day" // Unused, for backwards compatibility.
+			disabled={ isInvalid }
+			tabIndex={ isFocusable ? 0 : -1 }
+			aria-label={ getDayLabel( day, isSelected, numEvents ) }
+			column={ column }
+			isSelected={ isSelected }
+			isToday={ isToday }
+			hasEvents={ numEvents > 0 }
+			onClick={ onClick }
+			onKeyDown={ onKeyDown }
+		>
+			{ dateI18n( 'j', day, -day.getTimezoneOffset() ) }
+		</DayButton>
 	);
 }
 
+function getDayLabel( date: Date, isSelected: boolean, numEvents: number ) {
+	const { formats } = __experimentalGetSettings();
+	const localizedDate = dateI18n(
+		formats.date,
+		date,
+		-date.getTimezoneOffset()
+	);
```

Comment:
We could use `format` from `date-fns` but we would need to set up internationalisation which is a little tricky to do in a way that doesn't increase the size of the build. We'd have to pass the locale to the component which I'm not a huge fan of because it makes it difficult to switch `DatePicker` away from `date-fns` in the future.

https://date-fns.org/v2.16.1/docs/I18n

`dateI8n` from `@wordpress/date` already handles localisation and is configured properly by WordPress which is why I'm using it here.

My view on `@wordpress/date` is that it is _complimentary_ to a date util library (e.g. `date-fns`) and should only provide:

- A store for the WordPress site's date settings including format preference and timezone.
- A function similar to `wp_date` in PHP.

Everything else should be deprecated from `@wordpress/date` in favour of `date-fns`. Right now I believe that's just the `isInTheFuture` function.

Reply:
Gotcha. Not something that needs to be addressed urgently, but I'll see if I can open an issue on next steps for `@wordpress/date` (which should include what we discussed in this comment too)

Reply:
https://github.com/WordPress/gutenberg/issues/43533

Reply:
Thanks!